diff --git a/src/ethosu_device.h b/src/ethosu_device.h
index 02942b1..28506b8 100644
--- a/src/ethosu_device.h
+++ b/src/ethosu_device.h
@@ -59,12 +59,12 @@ struct ethosu_device
 /**
  * Initialize the device.
  */
-struct ethosu_device *ethosu_dev_init(void *const base_address, uint32_t secure_enable, uint32_t privilege_enable);
+enum ethosu_error_codes ethosu_dev_init(struct ethosu_device *dev);
 
 /**
  * Deinitialize the device.
  */
-void ethosu_dev_deinit(struct ethosu_device *dev);
+enum ethosu_error_codes ethosu_dev_deinit(struct ethosu_device *dev);
 
 /**
  * Initialize AXI settings for device.
diff --git a/src/ethosu_device_u55_u65.c b/src/ethosu_device_u55_u65.c
index 50b78af..22ea067 100644
--- a/src/ethosu_device_u55_u65.c
+++ b/src/ethosu_device_u55_u65.c
@@ -67,19 +67,8 @@ uint64_t __attribute__((weak)) ethosu_address_remap(uint64_t address, int index)
     return address;
 }
 
-struct ethosu_device *ethosu_dev_init(void *const base_address, uint32_t secure_enable, uint32_t privilege_enable)
+enum ethosu_error_codes ethosu_dev_init(struct ethosu_device *dev)
 {
-    struct ethosu_device *dev = malloc(sizeof(struct ethosu_device));
-    if (!dev)
-    {
-        LOG_ERR("Failed to allocate memory for Ethos-U device");
-        return NULL;
-    }
-
-    dev->reg        = (volatile struct NPU_REG *)base_address;
-    dev->secure     = secure_enable;
-    dev->privileged = privilege_enable;
-
 #ifdef ETHOSU55
     if (dev->reg->CONFIG.product != ETHOSU_PRODUCT_U55)
 #else
@@ -96,16 +85,15 @@ struct ethosu_device *ethosu_dev_init(void *const base_address, uint32_t secure_
         goto err;
     }
 
-    return dev;
+    return ETHOSU_SUCCESS;
 
 err:
-    free(dev);
-    return NULL;
+    return ETHOSU_GENERIC_FAILURE;
 }
 
-void ethosu_dev_deinit(struct ethosu_device *dev)
+enum ethosu_error_codes ethosu_dev_deinit(struct ethosu_device *dev)
 {
-    free(dev);
+    return ETHOSU_SUCCESS;
 }
 
 enum ethosu_error_codes ethosu_dev_axi_init(struct ethosu_device *dev)
