diff --git a/src/ethosu_device_u55_u65.c b/src/ethosu_device_u55_u65.c
index 22ea067..09dfc9e 100644
--- a/src/ethosu_device_u55_u65.c
+++ b/src/ethosu_device_u55_u65.c
@@ -69,16 +69,6 @@ uint64_t __attribute__((weak)) ethosu_address_remap(uint64_t address, int index)
 
 enum ethosu_error_codes ethosu_dev_init(struct ethosu_device *dev)
 {
-#ifdef ETHOSU55
-    if (dev->reg->CONFIG.product != ETHOSU_PRODUCT_U55)
-#else
-    if (dev->reg->CONFIG.product != ETHOSU_PRODUCT_U65)
-#endif
-    {
-        LOG_ERR("Failed to initialize device. Driver has not been compiled for this product");
-        goto err;
-    }
-
     // Make sure the NPU is in a known state
     if (ethosu_dev_soft_reset(dev) != ETHOSU_SUCCESS)
     {
